### PR TITLE
Feature - Add bitcoind/litecoind support to broker

### PR DESCRIPTION
**Pull request by dannypaz**
## Description
This adds litecoind/bitcoind support to the broker in the form of build commands.

    npm run build local - simnet setup w/ btcd/ltcd
    npm run build - user defined (user must edit their own .env files)

All currency networks, other than simnet, make use of bitcoind/litecoind.

**BREAKING CHANGE**:

This PR may have unintended consequences when a user upgrades. This PR changes the default currency daemons from btcd/ltcd to bitcoind/litecoind.

Tasks:
1. Backup user wallet
2. start bitcoind/litecoind and let them sync to testnet
3. Remove blocks directory from lnd
4. Start `docker-compose up -d`

More information will be posted in changelog

## Related PRs
https://github.com/sparkswap/lnd-engine/pull/102
https://github.com/sparkswap/lnd-engine/pull/100
https://github.com/sparkswap/relayer/pull/169
https://github.com/sparkswap/lnd/pull/15

## Todos
- [n/a] Tests
- [x] Documentation
- [x] Link to Trello

### CODE REVIEW

File: scripts/build.sh [hunk 1]
```diff
@@ -41,18 +41,23 @@ rm -rf ./proto/.git
...
-  cp ./.env-simnet-local-sample ./.env
+  cp ./.env-simnet-sample ./.env
   echo "Copying dev file to 'docker-compose.override.yml'"
-  cp ./docker-compose.dev.yml ./docker-compose.override.yml
-# If the build is not local AND the override exists for whatever reason,
-# we should delete the file to not mess w/ the build
+  cp ./docker-compose.simnet.yml ./docker-compose.override.yml
+elif [ "$ARG" == "regtest" ] || [ "$ARG" == "testnet" ] || [ $ARG == "mainnet" ]; then
+  echo "Copying env $ARG local file to .env"
+  cp ./.env-$ARG-sample ./.env
```
Comment (treygriffith) [inline scripts/build.sh]: I don't like copying sample files like this

File: scripts/build.sh [hunk 2]
```diff
@@ -41,18 +41,23 @@ rm -rf ./proto/.git
...
+  cp ./docker-compose.$ARG.yml ./docker-compose.override.yml
 elif [ -f docker-compose.override.yml ]; then
-  echo "Deleting local docker-compose.override.yml file"
-  rm -f ./docker-compose.override.yml
+  # Let the user know that an override file exists which may mean that the user
+  # will have settings they do not expect
+  echo "A 'docker-compose.override.yml' file exists, but you are not using a supported"
+  echo "environment: local or regtest"
+  echo ""
+  echo "WARNING: This may add unwanted settings to the relayer that could effect how your daemon runs."
```
Comment (treygriffith) [inline scripts/build.sh]: this is the broker, not the relayer

**Comment by treygriffith** [comment]
This is a pretty significant breaking change and for anyone that has a broker running will require pretty big changes (including re-syncing, etc).

Can we note this as a breaking change somewhere and include some migration documentation?

File: docker-compose.yml [hunk 3]
```diff
@@ -82,43 +84,40 @@ services:
...
+    image: sparkswap_bitcoind:latest
     volumes:
       - shared:/shared
       - bitcoin:/data
     environment:
       - RPC_USER=${BTC_RPC_USER}
       - RPC_PASS=${BTC_RPC_PASS}
       - NETWORK=${NETWORK}
-      - DEBUG=info
+      - DEBUG=debug
```
[the remark this answers is not in the document]
Comment (dannypaz) [reply docker-compose.yml]: All the DEBUGs were used for dev. Thanks for the catch

File: .env-mainnet-sample [hunk 4]
```diff
@@ -0,0 +1,60 @@
+###############################################################
+# MainNet Sparkswap Environment File
```
[the remark this answers is not in the document]
Comment (dannypaz) [reply .env-mainnet-sample]: Removing this and the build command

File: scripts/build.sh (hunk 1 above)
Comment (dannypaz) [reply scripts/build.sh]: @treygriffith I agree for the broker. We will still have `npm run build local` or `npm run build` and provide the sample files for users during setup. I will remove this line

File: .env-regtest-sample [hunk 5]
```diff
@@ -1,23 +1,23 @@
...
 #
-NETWORK=simnet
+NETWORK=regtest
 
-# Below are the urls that are used for the sparkswap daemon to be connected to a
+# Below are the urls that are used for sparkswapd to be connected to a
 # sparkswap hosted relayer.
 #
-# Relayer RPC Hosts for Local SimNet
+# Relayer RPC Hosts for SimNet
```
Comment (treygriffith) [inline .env-regtest-sample]: use a consistent name here

**Pull request merged**